### PR TITLE
make buttons pressable and focusable!

### PR DESCRIPTION
**Pull request by farooqkz**
This PR makes Buttons focus when their `isFocused` prop evaluates to true. Also when they are focused, if the user presses `Enter` key, the `onClick` callback is fired(if provided).

### CODE REVIEW

**Comment by Simon-Laux** [comment]
I think it would be better to register and unregister the event handler in "Class component lifecycle methods" (https://www.infernojs.org/docs/guides/components). I think `componentDidMount` and `componentWillUnmount` are the correct ones?

But generally I'm not sure about how to best do these keyboard navigation things. I'm also not sure about having focus state in the properties, what about if multiple elements do that? then have race condition bugs. 🤷

**Comment by farooqkz** [comment]
> I think it would be better to register and unregister the event handler in "Class component lifecycle methods" (https://www.infernojs.org/docs/guides/components). I think `componentDidMount` and `componentWillUnmount` are the correct ones?
> 
> But generally I'm not sure about how to best do these keyboard navigation things. I'm also not sure about having focus state in the properties, what about if multiple elements do that? then have race condition bugs. shrug

Good idea to use lifecycle methods. But in Inferno, to use them you don't need to use class components. Regarding the race condition thing, it is responsibility of the parent component to handle that and that only one component must have focus.

**Comment by farooqkz** [comment]
After some thought, I don't think it's necessary to remove the event handler on component unmount. Because it is tied to the component itself, it will get removed when it gets unmounted anyway. But I agree about not using inline function.

File: KaiUIngInferno/src/ui/Button.tsx [hunk 1]
```diff
@@ -34,7 +43,15 @@ export default function Button({ text, isFocused, onClick, icon, iconSrc, iconSi
...
   return (
-    <div className={buttonCls}>
+    <div onKeyDown={ linkEvent(props, handleKeyDown) } className={buttonCls} tabIndex={0} ref={(ref: HTMLElement | null) => {
+      if (ref) {
+        if (isFocused) {
+          ref.focus();
+        } else {
+          ref.blur();
+        }
+      }
```
Comment (Simon-Laux) [inline KaiUIngInferno/src/ui/Button.tsx:53]: I'm not sure when this is run? is it run when the component first enters the DOM? or is it run on every render call? I would prefer setting a class variable to the ref and running the is focused stuff in the correct inferno lifecycle function (I guess it's the one when props change?)

Comment (farooqkz) [reply KaiUIngInferno/src/ui/Button.tsx:53]: You are right it's not good for performance: https://github.com/infernojs/inferno/issues/1454

**Pull request merged**